tests/test_wizard.py

```diff
@@ -0,0 +1,249 @@
+        v_init = w.start()
+        w._daemon.network.reset()
+
+        d = {'autoconnect': True, 'want_proxy': True}
+        self.assertFalse(w.is_last_view(v_init.view, d))
+        v = w.resolve_next(v_init.view, d)
+        self.assertEqual('proxy_config', v.view)
+        self.assertEqual(True, self.config.NETWORK_AUTO_CONNECT)
+        d_proxy = {'enabled': True, 'mode': 'socks5', 'host': 'localhost', 'port': '1'}
+        d.update({'proxy': d_proxy})
+        v = w.resolve_next(v.view, d)
+        self.assertTrue(w.is_last_view(v.view, d))
+
+        self.assertTrue(w._daemon.network.run_called)
+        self.assertEqual(NetworkParameters(server=None, proxy=d_proxy, auto_connect=True, oneserver=None), w._daemon.network.parameters)
+
+    async def test_proxy_and_server(self):
+        w = ServerConnectWizard(DaemonMock(self.config))
+        v_init = w.start()
+        w._daemon.network.reset()
+
+        d = {'autoconnect': False, 'want_proxy': True}
+        self.assertFalse(w.is_last_view(v_init.view, d))
+        v = w.resolve_next(v_init.view, d)
+        self.assertEqual('proxy_config', v.view)
+        self.assertEqual(False, self.config.NETWORK_AUTO_CONNECT)
+        d_proxy = {'enabled': False}
+        d.update({'proxy': d_proxy})
+        v = w.resolve_next(v.view, d)
+
+        w._daemon.network.reset()
+        self.assertEqual('server_config', v.view)
+        d.update({'server': 'localhost:1:t'})
+        self.assertTrue(w.is_last_view(v.view, d))
+        v = w.resolve_next(v.view, d)
+
+        serverobj = ServerAddr.from_str_with_inference('localhost:1:t')
+        self.assertTrue(w._daemon.network.run_called)
+        self.assertEqual(NetworkParameters(server=serverobj, proxy=None, auto_connect=False, oneserver=None), w._daemon.network.parameters)
+
+
+class WalletWizardTestCase(WizardTestCase):
+
+    def wizard_for(self, *, name, wallet_type):
+        w = NewWalletWizard(DaemonMock(self.config), self.plugins)
+        if wallet_type == '2fa':
+            w.plugins.get_plugin('trustedcoin').extend_wizard(w)
+        v_init = w.start()
+        self.assertEqual('wallet_name', v_init.view)
+        d = {'wallet_name': name}
+        self.assertFalse(w.is_last_view(v_init.view, d))
+        v = w.resolve_next(v_init.view, d)
+        self.assertEqual('wallet_type', v.view)
+
+        d.update({'wallet_type': wallet_type})
+        w.resolve_next(v.view, d)
+
+        return w
+
+    async def test_create_standard_wallet_newseed(self):
+        w = self.wizard_for(name='test_standard_wallet', wallet_type='standard')
+        v = w._current
+        d = v.wizard_data
+        self.assertEqual('keystore_type', v.view)
+
+        d.update({'keystore_type': 'createseed'})
+        v = w.resolve_next(v.view, d)
+        self.assertEqual('create_seed', v.view)
+
+        d.update({'seed': '9dk', 'seed_type': 'segwit', 'seed_extend': False, 'seed_variant': 'electrum',
+                  'seed_extra_words': False})
```

Comment:
Just to clarify, these fields are populated by the GUIs and the base lib, that's why you need to set them all manually here, right?
E.g. at first glance I would have expected not to have to specify the "seed" in the "createseed" flow, but I guess that must be the reason.
So these tests check whether the fields are properly processed and test if "resolve_next" works, but they cannot test if population of the fields is working?

Reply:
> Just to clarify, these fields are populated by the GUIs and _not_ the base lib, that's why you need to set them all manually here, right? E.g. at first glance I would have expected not to have to specify the "seed" in the "createseed" flow, but I guess that must be the reason. So these tests check whether the fields are properly processed and test if "resolve_next" works, but they cannot test if population of the fields is working?

Correct. To test populating the wizard_data, we should create testcases that use the UI classes.

The testcases in this commit are simply testing the wizard flow.